src/main/kotlin/ru/meanmail/codeInspection/InstalledPackageInspection.kt

```diff
@@ -34,52 +36,49 @@ class InstalledPackageInspection : LocalInspectionTool() {
                 }
 
                 val packageName = element.name.text ?: return
-                val versionOneList = element.versionspec?.versionMany?.versionOneList
-                val version = versionOneList?.get(0)?.version?.text
-                val task = getVersionAsync(element.project, packageName, version)
-                val versions = task.get() ?: return
+                val versionSpec = element.versionspec
+                val requirement = PyRequirementParser.fromLine(element.text)
 
-                val required = versions.first
-                val installed = versions.second
-                val latest = versions.third
+                val task = getVersionAsync(element.project, packageName)
+                val versions = task.get() ?: return
+                val installed = versions.first
+                val latest = versions.second
 
                 val versionsRepresentation =
-                        "required: ${required?.presentableText ?: "<unknown>"}, " +
-                                "installed: ${installed?.presentableText ?: "<nothing>"}, " +
+                        "required: ${versionSpec?.text ?: "<unknown>"}, " +
+                                "installed: ${installed?.version ?: "<nothing>"}, " +
                                 "latest: ${latest?.presentableText ?: "<unknown>"}"
 
 
-                if (required != null && required != installed) {
-                    val message = "'$packageName' version '${required.presentableText}' " +
+                if (requirement != null && (installed == null || !installed.matches(requirement))) {
+                    val message = "'${requirement.presentableText}' " +
                             "is not installed ($versionsRepresentation)"
                     holder.registerProblem(element,
                             message,
                             InstallPackageQuickFix(element,
-                                    "Install '$packageName' " +
-                                            "version '${required.presentableText}' " +
+                                    "Install '${requirement.presentableText}' " +
                                             "($versionsRepresentation)",
-                                    required.presentableText))
+                                    requirement))
                 }
 
-                if (latest != null && required != null && required < latest) {
-                    val message = "'$packageName' version '${required.presentableText}' " +
+                if (latest != null && installed != null && PyPackageVersionNormalizer.normalize(installed.version) < latest) {
+                    val message = "'$packageName' version '${installed.version}' " +
```

Comment:
Now the message `outdated` always appears along with `is not installed`. It should not be.

Should be something like this:

```kotlin
val latest: Package?

...

if (latest != null && requirement.match(listOf(latest)) == null) {
    ...
}
```

Reply:
I fixed it by just adding an `else`